### PR TITLE
LPS-183476 | Automations for Date Range filters for the Dataset View

### PR DESCRIPTION
**Pull request by JerryAdrianoo**
**Description:**
Add automations for Date Range filters for the Dataset View

**JIRA Ticket:**
https://issues.liferay.com/browse/LPS-183476

**OBS:** 
Resend this [PR](https://github.com/liferay-frontend/liferay-portal/pull/3384) because it has broken for some reason.

### CODE REVIEW

**Comment by JerryAdrianoo** [comment]
ci:test:sf

**Comment by JerryAdrianoo** [comment]
ci:test:sf

**Comment by JerryAdrianoo** [comment]
ci:test:frontend-data-set

**Comment by JerryAdrianoo** [comment]
ci:test:sf

**Comment by JerryAdrianoo** [comment]
ci:test:sf

**Comment by JerryAdrianoo** [comment]
ci:test:sf

**Comment by JerryAdrianoo** [comment]
cc: @CarlosBrichete

**Comment by JerryAdrianoo** [comment]
ci:test:frontend-data-set

**Comment by JerryAdrianoo** [comment]
ci:test:frontend-data-set

**Comment by JerryAdrianoo** [comment]
ci:test:sf

File: modules/apps/frontend-data-set/frontend-data-set-test/src/testFunctional/macros/FrontendDataSetAdmin.macro [hunk 1]
```diff
@@ -159,6 +167,36 @@ definition {
...
+
+		Select(
+			locator1 = "FrontendDataSetAdmin#FILTER_BY_SELECT",
+			value1 = ${key_filterType});
+
+		Click(
+			key_text = "Save",
+			locator1 = "CP2ActivationKeys#KEY_DEACTIVATION_TERMS");
+	}
+
```
Comment (CarlosBrichete) [inline modules/apps/frontend-data-set/frontend-data-set-test/src/testFunctional/macros/FrontendDataSetAdmin.macro:199]: This macro is not needed, you can just use createMultipleFilters() and pass just one parameter to it. Also, as you will use just this macro, I would rename it to createFilters()

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/frontenddataset/datamanagment/DatasetsFilters.testcase [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+		}
+
+		task ("asserts all fields in the schema are displayed") {
+			Click(locator1 = "FrontendDataSetAdmin#FILTER_BY_SELECT");
+
+			FrontendDataSetAdmin.assertSearchFilters(key_filterList = "creator,externalReferenceCode,id,status,createDate,fieldName,from,modifiedDate,name,r_fdsViewFDSDateFilterRelationship_c_fdsViewId,rendererType,r_fdsViewFDSDateFilterRelationship_c_fdsViewERC,to,type");
+		}
+	}
+
+	@description = "LPS-183476. Confirm that when 'Name Field' is not provided, it is filled with the same name as field name."
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/frontenddataset/datamanagment/DatasetsFilters.testcase:68]: Put the correct LPS--> LPS-176184 (in Poshi, we should put the LPS of the original story/bug)

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/frontenddataset/datamanagment/DatasetsFilters.testcase [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+		task ("Then Confirm that the name of the saved filter is the same name as the chosen field") {
+			AssertElementPresent(
+				chosenFilter = "id",
+				key_tableColumn = 1,
+				key_tableRow = 1,
+				locator1 = "FrontendDataSetAdmin#FILTER_TABLE_CELL");
+		}
+	}
+
+	@description = "LPS-183476. Confirm that 'No default filters were created.' and 'Start creating a filter to display specific data' messages are displayed"
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/frontenddataset/datamanagment/DatasetsFilters.testcase:92]: Put the correct LPS--> LPS-176184 (in Poshi, we should put the LPS of the original story/bug)

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/frontenddataset/datamanagment/DatasetsFilters.testcase [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+				locator1 = "Message#EMPTY_STATE_INFO",
+				value1 = "No default filters were created.");
+
+			AssertTextEquals.assertText(
+				locator1 = "Message#EMPTY_STATE_TEXT",
+				value1 = "Start creating a Filter to display specific Data.");
+		}
+	}
+
+	@description = "LPS-183476. Confirm that the New filter modal is closed and the changes are not applied when canceling on Cancel button"
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/frontenddataset/datamanagment/DatasetsFilters.testcase:106]: Put the correct LPS--> LPS-176184 (in Poshi, we should put the LPS of the original story/bug)

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/frontenddataset/datamanagment/DatasetsFilters.testcase [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+				key_modalTitle = "New Filter",
+				locator1 = "Button#CLOSE_MODAL");
+
+			AssertTextEquals.assertText(
+				locator1 = "Message#EMPTY_STATE_INFO",
+				value1 = "No default filters were created.");
+		}
+	}
+
+	@description = "LPS-183476. Confirm that the New filter modal is closed and the changes are not applied when canceling on Close (X) button"
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/frontenddataset/datamanagment/DatasetsFilters.testcase:128]: Put the correct LPS--> LPS-176184 (in Poshi, we should put the LPS of the original story/bug)

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/frontenddataset/datamanagment/DatasetsFilters.testcase [hunk 6]
```diff
@@ -0,0 +1,214 @@
...
+				key_modalTitle = "New Filter",
+				locator1 = "Button#CLOSE_MODAL");
+
+			AssertTextEquals.assertText(
+				locator1 = "Message#EMPTY_STATE_INFO",
+				value1 = "No default filters were created.");
+		}
+	}
+
+	@description = "LPS-183476. Confirm that the text 'Select' is labeling the  'Filter By Field'"
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/frontenddataset/datamanagment/DatasetsFilters.testcase:152]: Put the correct LPS--> LPS-176184 (in Poshi, we should put the LPS of the original story/bug)

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/frontenddataset/datamanagment/DatasetsFilters.testcase [hunk 7]
```diff
@@ -0,0 +1,214 @@
...
+		task ("Then asserts the selected option is actually displayed") {
+			AssertElementPresent(
+				chosenFilter = "id",
+				key_tableColumn = 1,
+				key_tableRow = 1,
+				locator1 = "FrontendDataSetAdmin#FILTER_TABLE_CELL");
+		}
+	}
+
+	@description = "LPS-183476. Confirm that 'Filter By' field cannot be empty"
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/frontenddataset/datamanagment/DatasetsFilters.testcase:170]: Put the correct LPS--> LPS-176184 (in Poshi, we should put the LPS of the original story/bug)

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/frontenddataset/datamanagment/DatasetsFilters.testcase [hunk 8]
```diff
@@ -0,0 +1,214 @@
...
+		}
+
+		task ("Then Comfirm that is not possible clicks save button") {
+			AssertElementPresent(
+				key_text = "Save",
+				locator1 = "Button#ANY_DISABLED");
+		}
+	}
+
+	@description = "LPS-183476. Confirm that is possible search filter when click on search icon"
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/frontenddataset/datamanagment/DatasetsFilters.testcase:190]: Put the correct LPS--> LPS-176184 (in Poshi, we should put the LPS of the original story/bug)

**Review by CarlosBrichete — CHANGES_REQUESTED** [review]

**Review by CarlosBrichete — APPROVED** [review]

**Comment by CarlosBrichete** [comment]
LGTM :)

**Pull request closed without merging**